core-api/src/main/java/com/optimizely/ab/config/Experiment.java

```diff
@@ -173,6 +173,101 @@ public boolean isLaunched() {
         return status.equals(ExperimentStatus.LAUNCHED.toString());
     }
 
+    public String serializeConditions(Map<String, String> audiencesMap) {
+        Condition condition = this.audienceConditions;
+        return condition instanceof EmptyCondition ? "" : this.serialize(condition, audiencesMap);
+    }
+
+    private String getNameFromAudienceId(String audienceId, Map<String, String> audiencesMap) {
+        if (audiencesMap == null ) return "\"" + audienceId + "\"";
+        String audienceName = "\"" + audiencesMap.get(audienceId) + "\"";
+        return audiencesMap.get(audienceId) != null ? audienceName : "\"" + audienceId + "\"";
+    }
+
+    private String getOperandOrAudienceId(Condition condition, Map<String, String> audiencesMap){
```

Comment:
all belong to same base class, can't we add one method in Conditions<T> that should return the appropriate String, instead of all these conditions.
@jaeopt wdyt?

Reply:
I like this! Let me try and update.